### DISCUSSION TITLE
How to get minimalized segmentation

### DISCUSSION THREAD

**Discussion by YunusEmreSayin** [open-edge-platform/anomalib#2271] [Q&A] [answered]
I can detect the anomaly on fabric. But the are is so large in segmentation result. How can I minimalize the segmentation? Like now I have segmentation that contains even light yellow in the heat map. Maybe its better to contains just the orange or dark yellow part of heatmap. I am waiting for solutions for it please
This one is current:
![combined_raw_images_49](https://github.com/user-attachments/assets/5c2552fa-74d3-47ec-8b3e-2efb73fd5203)

And I want a segmentation like this maybe:
![combined_raw_images_49](https://github.com/user-attachments/assets/0964a0b3-f1df-447a-86a5-1676abc19a57)

**Comment by abc-125** [comment]
You can try to set a higher threshold manually, so only higher values (warmer colors on the heatmap) will be included into segmentation, change `default_value` below:

```
from anomalib.metrics import ManualThreshold
from anomalib.engine import Engine

manual_threshold = ManualThreshold(default_value=0.5)
engine = Engine(threshold=manual_threshold)
```

**Reply by YunusEmreSayin** [reply]
I was already giving the threshold value manually. But its so sensitive and I wanna mark the hot colors only. Is there any parameter or way for this?

**Reply by abc-125** [reply]
I don't think so, you can try another anomaly detection model or use more good images for training, so you have less false positives.

**Reply by samet-akcay** [reply, marked as answer]
As @abc-125 mentioned, you could try a different model that suits your needs. For the existing model you are using, you could check the `AnomalyMapGenerator` module, and post-processing steps. For example, this is the post-processing steps for PatchCore.

https://github.com/openvinotoolkit/anomalib/blob/b7188db4e49c8443a531d932927395807d52b1a6/src/anomalib/models/image/patchcore/anomaly_map.py#L13-L28

`sigma` is a parameter that defines the gaussian kernel size, which smooths the heatmap. You could play with this parameter to see if a smaller `sigma` gives you smaller detection output
